crates/stages/src/stages/tx_lookup.rs

```diff
@@ -66,14 +66,35 @@ impl<DB: Database> Stage<DB> for TransactionLookupStage {
                 .unwrap_or_default()
         });
 
-        // Collect tranasctions for each body and insert the reverse lookup for hash -> tx_id.
+        // Collect transactions for each body
+        let mut tx_list = vec![];
         for body_entry in bodies {
             let (_, body) = body_entry?;
-            let transactions = tx_cursor.walk(body.start_tx_id)?.take(body.tx_count as usize);
+            let transactions = cursor_tx.walk(body.start_tx_id)?.take(body.tx_count as usize);
 
             for tx_entry in transactions {
                 let (id, transaction) = tx_entry?;
-                tx.put::<tables::TxHashNumber>(transaction.hash(), id)?;
+                tx_list.push((transaction.hash(), id));
+            }
+        }
+
+        let mut cursor_txhash = tx.cursor_write::<tables::TxHashNumber>()?;
+
+        // If the last element is smaller than our first, then we can just append into the DB.
+        // This probably only ever happens during sync, on the first table insertion.
+        let last = cursor_txhash.last()?;
+        let mut append = last.is_none();
+        if let (false, Some((last_txhash, _))) = (tx_list.is_empty(), last) {
+            append = last_txhash < tx_list[0].0;
```

Comment:
We are comparing it with the first in the list, should `tx_list` be sorted at this pont?